documentation/how-to/scaling.md

```diff
@@ -0,0 +1,357 @@
+The Temporal Server consists of four independently scalable services:
+
+- Frontend gateway: for rate limiting, routing, authorizing.
+- History subsystem: maintains data (mutable state, queues, and timers).
+- Matching subsystem: hosts Task Queues for dispatching.
+- Worker Service: for internal background Workflows.
+
+For example, a real-life production deployment can have 5 Frontend, 15 History,
+17 Matching, and 3 Worker Services per cluster.
+
+The Temporal Server services can run independently or be grouped together into
+shared processes on one or more physical or virtual machines. For live
+(production) environments, we recommend that each service runs independently,
+because each one has different scaling requirements and troubleshooting becomes
+easier. The History, Matching, and Worker Services can scale horizontally within
+a Cluster. The Frontend Service scales differently than the others because it
+has no sharding or partitioning; it is just stateless.
+
+(The above was extracted from
+[Temporal's documentation](https://docs.temporal.io/clusters#temporal-server))
+
+The Temporal server charm is designed such that each service can be deployed as
+a separate application which can then be connected together by integrating with
+the same database. These services can then be scaled according to your needs.
+
+To deploy the services separately in a scalable way, you must deploy the
+application as follows:
+
+```bash
+juju deploy temporal-k8s --config services="frontend"
```

Comment:
I thought we only had the UI and worker charms for Temporal. What is `temporal-k8s-history` for example? I am asking for myself.

Reply:
Sorry for the confusion! I just updated the instructions as it should have been ```juju deploy temporal-k8s --config services="history" temporal-k8s-history```, meaning it will deploy the ```temporal-k8s``` charm with the name ```temporal-k8s-history``` and assign the services config value to be "history" only.

Reply:
Ah was wondering something similar, very cool the way you can use the same charm to spin up the individual components.